test/message/esm_display_syntax_error.mjs

```diff
@@ -1,2 +1,4 @@
+// Flags: --no-harmony-top-level-await
```

Comment:
@devsnek is this sufficient for you for a `no-top-level-await` test?

Reply:
no. what needs to be tested is that node does not forward the result of Module::Evaluate() when TLA is disabled. The test case is that the last statement in a module returns a rejected promise.

```js
// Flags: --no-harmony-top-level-await

process.on('unhandledRejection', common.mustCall());
Promise.reject(new Error('should not be fatal error')); // last line of file
```

Reply:
We should probably have two versions of this test, one with `Flags: --no-harmony-top-level-await` and one without.

Reply:
Does it need a `setTimeout` or another tick mechanism to force the handler to actually run?

Reply:
@jkrems no

Reply:
This test is failing
```
Error: should not be fatal error
    at Object.<anonymous> (/Users/mylesborins/code/node/main/test/parallel/test-no-harmony-top-level-await.js:6:16)
```

Reply:
@MylesBorins well it definitely shouldn't fail in a .js file, regardless of the flags. the tests should be .mjs though.

Reply:
@devsnek it is failing as a .js file and as a .mjs, on both master and this feature branch. It would appear the importing `common/index.mjs` is what is causing the failure to happen. I'll dig into the common library again to see if we need to do something extra

Reply:
Ok got it... PTAL